apps/demo/src/app/shared/department-children/update-id-function.ts

```diff
@@ -9,10 +9,11 @@ export function updateId(
   idName = 'id',
 ): DepartmentChild[] {
   return rows.map((row) => {
+    // convert the row to a record so we can access the idName
     const itemRecord = castTo<Record<string, string>>(row);
-    return castTo<DepartmentChild>({
+    return {
```

Comment:
row as Record<string, string> produces this error:

"Conversion of type 'DepartmentChild' to type 'Record<string, string>' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first."

The recommendation in the error message is exactly what castTo() does.